### PR TITLE
refactor(nasha): spread task tracker component

### PR DESCRIPTION
**Pull request by y4nnL**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/nasha-refresh-accesses-tab`
| Bug fix?         | yes
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-9549
| License          | BSD 3-Clause

- [ ] ~~Try to keep pull requests small so they can be easily reviewed.~~
- [X] Commits are signed-off
- [X] Only FR translations have been updated
- [X] Branch is up-to-date with target branch
- [X] Lint has passed locally
- [X] Standalone app was ran and tested locally
- [X] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

### CODE REVIEW

**Review by antleblanc — CHANGES_REQUESTED** [review]
@y4nnL Could you please resolve existing conflicts?
Thank you!

File: packages/manager/modules/nasha/README.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- Checker les components bindings / resolve de chaque module
```
[the remark this answers is not in the document]
Comment (y4nnL) [reply packages/manager/modules/nasha/README.md:1]: :+1: my bad...
I've cleaned this file in the "ultimate" PR : https://github.com/ovh/manager/pull/7495/files#diff-d47fe37417d7c525bcf8b9aef8630b63e1cb822bca76a0acf023dd106114da35

File: packages/manager/modules/nasha/README.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+- bug si pas de customName sur le service => mettre le nom du service
```
Comment (ghyenne) [inline packages/manager/modules/nasha/README.md:2]: Maybe translate in english.

File: packages/manager/modules/nasha/src/components/partition/create/create.constants.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export const TRANSLATE = 'nasha_components_partition_create';
```
Comment (ghyenne) [inline packages/manager/modules/nasha/src/components/partition/create/create.constants.js:1]: More understanding

```suggestion
export const TRANSLATE_PREFIX = 'nasha_components_partition_create';
```

File: packages/manager/modules/nasha/src/components/partition/zfs-options/zfs-options.constants.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  options: /* @ngInject */ ($q, $http, partitionApiUrl, prepareZfsOptions) =>
+    $http
+      .get(`${partitionApiUrl}/options`)
+      .catch((error) =>
+        error.status === 404 ? { data: prepareZfsOptions() } : $q.reject(error),
+      )
+      .then(({ data }) => prepareZfsOptions(data)),
+};
+
+export const TRANSLATE = 'nasha_components_partition_zfs_options';
```
Comment (ghyenne) [inline packages/manager/modules/nasha/src/components/partition/zfs-options/zfs-options.constants.js:11]:
```suggestion
export const TRANSLATE_PREFIX = 'nasha_components_partition_zfs_options';
```

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.constants.js [hunk 5]
```diff
@@ -1,11 +1,9 @@
-export const CUSTOM_SNAPSHOT_NAME_PREFIX = 'snap';
-export const CUSTOM_SNAPSHOT_NAME_SEPARATOR = '-';
 export const STATE_NAME = 'nasha.dashboard.partition.accesses';
+export const TASK_TRACKER_STATE_NAME = `${STATE_NAME}.task-tracker`;
 export const TRANSLATE = 'nasha_dashboard_partition_accesses';
```
Comment (ghyenne) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.constants.js:3]:
```suggestion
export const TRANSLATE_PREFIX = 'nasha_dashboard_partition_accesses';
```

**Review by ghyenne — COMMENTED** [review]
More understanding if replace 'TRANSLATE' with 'TRANSLATE_PREFIX'

File: packages/manager/modules/nasha/README.md (hunk 2 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/README.md:2]: :+1: 
Removed here : https://github.com/ovh/manager/pull/7495/files#diff-d47fe37417d7c525bcf8b9aef8630b63e1cb822bca76a0acf023dd106114da35L1-L2

File: packages/manager/modules/nasha/src/components/partition/create/create.constants.js (hunk 3 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/components/partition/create/create.constants.js:1]: :+1: 
This has been done here : https://github.com/ovh/manager/pull/7495/files#diff-11d14b2a9fc39bc7fa2a726a25d15dda322ebacbe416ed3e98718738340c92c8R1

File: packages/manager/modules/nasha/src/components/partition/zfs-options/zfs-options.constants.js (hunk 4 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/components/partition/zfs-options/zfs-options.constants.js:11]: :+1: 
This has been done here : https://github.com/ovh/manager/pull/7495/files#diff-7915a2ac644befb82d1c7933cf78c5f0261701d9a5638cdfee26494028c46ea9R1

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.constants.js (hunk 5 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.constants.js:3]: :+1: 
Removed : https://github.com/ovh/manager/pull/7495/files#diff-4ba66823bb79f831097d103cc4a5bdd67e8fd97c9aa698c60963c3ae4373effb

File: packages/manager/modules/nasha/src/dashboard/partition/snapshots/snapshots.constants.js [hunk 6]
```diff
@@ -1,11 +1,13 @@
 export const CUSTOM_SNAPSHOT_NAME_PREFIX = 'snap';
 export const CUSTOM_SNAPSHOT_NAME_SEPARATOR = '-';
 export const STATE_NAME = 'nasha.dashboard.partition.snapshots';
+export const TASK_TRACKER_STATE_NAME = `${STATE_NAME}.task-tracker`;
 export const TRANSLATE = 'nasha_dashboard_partition_snapshots';
```
[the remark this answers is not in the document]
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/snapshots/snapshots.constants.js:5]: :+1: 
This has been done here : https://github.com/ovh/manager/pull/7495/files#diff-f30ea6ace87b49cc69a23608317d821378113ec0b2f4f43abebfe3228f682b07R3

**Review by ghyenne — APPROVED** [review]

File: packages/manager/modules/nasha/src/dashboard/partition/partition.routing.js [hunk 7]
```diff
@@ -65,27 +71,27 @@ export default /* @ngInject */ ($stateProvider) => {
...
-      ) =>
-        iceberg(`/dedicated/nasha/${serviceName}/task`)
-          .query()
-          .expand('CachedObjectList-Pages')
-          .addFilter('status', 'in', Object.values(NashaTask.status))
-          .addFilter('partitionName', 'eq', partitionName)
-          .execute(null, true)
-          .$promise.then(({ data: tasks }) => prepareTasks(tasks)),
+      trackTasks: /* @ngInject */ ($state) => (params) =>
+        $state.go(TASK_TRACKER_STATE_NAME, params),
```
Comment (anooparveti) [inline packages/manager/modules/nasha/src/dashboard/partition/partition.routing.js:81]: **nitpick:** function name suggests it does something related to tracking while it does a state navigation. Can you rename according to its responsibility?

File: packages/manager/modules/nasha/src/dashboard/partition/snapshots/snapshots.routing.js [hunk 8]
```diff
@@ -1,45 +1,40 @@
...
-          .then(({ data: snapshots }) =>
-            prepareSnapshots(snapshots, customSnapshots, snapshotEnum),
-          ),
-      snapshotEnum: /* @ngInject */ (schema) =>
-        schema.models[snapshotEnumKey].enum,
+      snapshotTypes: /* @ngInject */ ($http, partitionApiUrl) =>
+        $http.get(`${partitionApiUrl}/snapshot`).then(({ data }) => data),
       trackTasks: /* @ngInject */ ($state) => (params) =>
-        $state.go(`${STATE_NAME}.task-tracker`, params),
+        $state.go(TASK_TRACKER_STATE_NAME, params),
```
Comment (anooparveti) [inline packages/manager/modules/nasha/src/dashboard/partition/snapshots/snapshots.routing.js:32]: **nitpick**: function name suggests it does something related to tracking while it does a state navigation. Can you rename according to its responsibility?

File: packages/manager/modules/nasha/src/dashboard/partitions/partitions.routing.js [hunk 9]
```diff
@@ -1,48 +1,64 @@
...
+        partitionName,
+        error,
+      } = {}) =>
+        tasks
+          ? trackTasks({ tasks, partitionName })
+          : goBack({ stateName: STATE_NAME, error }),
+      partitionsHref: /* @ngInject */ ($state, serviceName) => () =>
+        $state.href(STATE_NAME, { serviceName }),
+      trackTasks: /* @ngInject */ ($state) => (params) =>
+        $state.go(TASK_TRACKER_STATE_NAME, params),
```
Comment (anooparveti) [inline packages/manager/modules/nasha/src/dashboard/partitions/partitions.routing.js:61]: **nitpick**: function name suggests it does something related to tracking while it does a state navigation. Can you rename according to its responsibility?

**Review by anooparveti — COMMENTED** [review]
https://github.com/ovh/manager/pull/7494#discussion_r928529039
https://github.com/ovh/manager/pull/7494#discussion_r928527304
https://github.com/ovh/manager/pull/7494#discussion_r928528327

**Review by anooparveti — APPROVED** [review]

**Pull request merged**